bikeshare_app/views.py

```diff
@@ -15,12 +16,14 @@ def index():
 @app.route("/admin")
 def admin():
     """The admin page"""
+    app.logger.debug('Rendering admin page')
     #TODO Check user credentials for admin role
     # Allow user through if check passed, else log attempt and return home
     return render_template('admin.html')
 
 # Bike request endpoint
 @app.route("/request_bike/<user_email>/<int:bike_id>")
-def request_bike():
+def request_bike(user_email, bike_id):
     """This endpoint is for requesting a bike"""
+    app.logger.debug('User %s requesting bike %d' % (user_email, bike_id))
```

Comment:
%-formatting is "old style formatting." The Python community is still trying to oust this in favor of the .format() method, eg 'User {} requesting bike {}'.format(user_email, bike_id). The {} is where the parameters will fall into and you can do lots of things with it, in fact the CFG that describes this new format string can be found [here](https://docs.python.org/3/library/string.html#format-string-syntax).